datafusion/sql/src/planner.rs

```diff
@@ -1882,6 +1882,14 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 self.sql_expr_to_logical_expr(*expr, schema, ctes)?,
             ))),
 
+            SQLExpr::IsTrue(expr) => Ok(Expr::IsTrue(Box::new(
+                self.sql_expr_to_logical_expr(*expr, schema, ctes)?,
+            ))),
```

Comment:
I think you might be able to avoid having to extend `Expr` and add a `PhysicalExpr` at all if you rewrote `IS TRUE` in the sql planner to `IS NOT DISTINCT FROM`

Something like

```suggestion
            SQLExpr::IsTrue(expr) => Ok(Expr::BinaryExpr {
                left: Box::new(self.sql_expr_to_logical_expr(*expr, schema, ctes)?),
                op: Operator::IsNotDistinctFrom,
                right: Box::new(lit(true)),
            }),
```

That way a query like `SELECT x IS TRUE` becomes `SELECT x IS NOT DISTINCT FROM TRUE`


Here is an example from postgres showing they are equivalent:

```sql
alamb=# select column1 is true, column1 is not distinct from true from (values (true), (false), (null)) as sq;
 ?column? | ?column? 
----------+----------
 t        | t
 f        | f
 f        | f
(3 rows)
```

The same type of transformation applies to `IS FALSE` 

Then this PR would likely be 8 lines of code and then the sql level tests

Reply:
My preference would be to keep the logical plan as close to the original query as possible. Different query engines will have different approaches for how to implement this in the physical plan. I also think that it is better UX for the user to see IsTrue/False in the logical plan if that's what their query contained.

I think it is a good idea for DataFusion to map the logical IsTrue /False to the physical expression IsDistinctFrom and that removes the need for the new physical expressions here.

Reply:
Okay, I think that makes sense to me. The only thing I was wondering was how this would handle non-boolean/non-null inputs, since we should error for all other datatypes?

Reply:
Yes, that is related to the work I am starting in https://github.com/apache/arrow-datafusion/pull/3222. I think we need to do some type validation here so that if we see `<expr> IS TRUE` where `expr` is not boolean, then we either add a cast or throw an error if it cannot be cast to boolean.